File: Python/1418. Display Table of Food Orders in a Restaurant.py
class Solution:
    def displayTable(self, orders: List[List[str]]) -> List[List[str]]:
        table_map = defaultdict(lambda: defaultdict(int))
        dish_set = set()

        for order in orders:
            table_number, dish = int(order[1]), order[2]
            dish_set.add(dish)
            table_map[table_number][dish] += 1

        result = []

        # Prepare headings
        headings = ["Table"] + sorted(dish_set)
        result.append(headings)

        # Fill the table
        for table_number, dish_count in sorted(table_map.items()):
            row = [str(table_number)]
            for dish in sorted(dish_set):
                row.append(str(dish_count.get(dish, 0)))
            result.append(row)

        return result

File: C++/1418. Display Table of Food Orders in a Restaurant.cpp
#include<bits/stdc++.h>
using namespace std;


class Solution {
public:
    vector<vector<string>> displayTable(vector<vector<string>>& orders) {
        map<int, map<string, int>> tableMap;
        set<string> dishSet;

        for (const vector<string>& order : orders) {
            int tableNumber = stoi(order[1]);
            string dish = order[2];

            dishSet.insert(dish);
            tableMap[tableNumber][dish]++;
        }

        vector<vector<string>> result;

        // Prepare headings
        vector<string> headings = { "Table" };
        for (const string& dish : dishSet) {
            headings.push_back(dish);
        }
        result.push_back(headings);

        // Fill the table
        for (const auto& entry : tableMap) {
            int tableNumber = entry.first;
            const map<string, int>& dishCount = entry.second;

            vector<string> row = { to_string(tableNumber) };
            for (const string& dish : dishSet) {
                if (dishCount.count(dish)) {
                    row.push_back(to_string(dishCount.at(dish)));
                }
                else {
                    row.push_back("0");
                }
            }
            result.push_back(row);
        }

        return result;
    }
};